### PR TITLE
hicops: new package

### PR DESCRIPTION
**Pull request by NessieCanCode**

### CODE REVIEW

**Review by NessieCanCode — COMMENTED** [review]
formatting corrections

**Review by NessieCanCode — COMMENTED** [review]
formatting corrections

File: var/spack/repos/builtin/packages/hicops/package.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Hicops(Package):
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/hicops/package.py]: You can derive from a specialized class to use a specific build-system:
```suggestion
class Hicops(CMakePackage):
```

File: var/spack/repos/builtin/packages/hicops/package.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    homepage = "https://hicops.github.io/index"
+    git = "https://github.com/hicops/hicops.git"
+    maintainers = ['pcdslab', 'mhaseeb123', 'nessiecancode']
+
+    version('release', branch='release')
+    version('develop', branch='develop')
+
+    # Build Options
+    variant('USE_MPI', default='ON',
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/hicops/package.py]: Can you use lowercase for variants? That's what all the other packages do. In this specific case you need an `mpi` Boolean variant. Look for instance at the `mpi` variant in `hdf5`.

File: var/spack/repos/builtin/packages/hicops/package.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+            description='CMake Option: Enable MPI support.',
+            values=('ON', 'OFF'))
+    variant('USE_TIMEMORY', default='OFF',
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/hicops/package.py]: We don't map 1:1 the names that the build-system expects. In this case you can use e.g. `timemory` and a variant name.

File: var/spack/repos/builtin/packages/hicops/package.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    variant('QCHUNK', default='10000',
+            description='CMake Option: Max size of each batch extracted from the '
+                        'dataset.')
+    variant('MAX_HYPERSCORE', default='100',
+            description='CMake Option: Maximum allowed hyperscore computed.')
+
+    # Build Controls
+    variant('CMAKE_BUILD_TYPE', default='RelWithDebInfo',
+            description='CMake Build Control: Build type.',
+            values=('Release', 'Debug', 'MinSizeRel', 'RelWithDebInfo'))
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/hicops/package.py]: There is no need to set this if you inherit from `CMakePackage`

File: var/spack/repos/builtin/packages/hicops/package.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    depends_on('git', type='build', when='@develop')
+    depends_on('boost')
+    depends_on('py-numpy')
+    depends_on('py-python-dateutil')
+    depends_on('py-setuptools')
+    depends_on('py-bottleneck')
+    depends_on('py-pyparsing')
+    depends_on('py-subprocess32')
+    depends_on('py-six')
+    depends_on('cmake@3.11:3.21.2')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/hicops/package.py]: This dependency is of type `build`.

File: var/spack/repos/builtin/packages/hicops/package.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    depends_on('py-setuptools-scm')
+    depends_on('pkgconf')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/hicops/package.py]: Same as `cmake`

**Review by alalazo — CHANGES_REQUESTED** [review]
I left an initial review with some required changes.

File: var/spack/repos/builtin/packages/hicops/package.py (hunk 1 above)
Comment (mhaseeb123) [reply var/spack/repos/builtin/packages/hicops/package.py]: Hi,
Just a minor comment but the 'Hicops' package name can stay in lowercase like in your original commit for consistency with other spack packages.

File: var/spack/repos/builtin/packages/hicops/package.py (hunk 3 above)
Comment (mhaseeb123) [reply var/spack/repos/builtin/packages/hicops/package.py]: This variant should also add the `timemory` package as a dependency.

**Review by mhaseeb123 — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/hicops/package.py (hunk 1 above)
Comment (NessieCanCode) [reply var/spack/repos/builtin/packages/hicops/package.py]: @mhaseeb123 it has to be CapWords convention to be pep8 compliant and pass the styling check within this repo.

Comment (mhaseeb123) [reply var/spack/repos/builtin/packages/hicops/package.py]: @NessieCanCode thank you for the clarification. As long as the package name remains all lowercased, all is well.

File: var/spack/repos/builtin/packages/hicops/package.py (hunk 6 above)
Comment (NessieCanCode) [reply var/spack/repos/builtin/packages/hicops/package.py]: done

File: var/spack/repos/builtin/packages/hicops/package.py (hunk 5 above)
Comment (NessieCanCode) [reply var/spack/repos/builtin/packages/hicops/package.py]: done

File: var/spack/repos/builtin/packages/hicops/package.py (hunk 4 above)
Comment (NessieCanCode) [reply var/spack/repos/builtin/packages/hicops/package.py]: It has been removed.

File: var/spack/repos/builtin/packages/hicops/package.py (hunk 3 above)
Comment (NessieCanCode) [reply var/spack/repos/builtin/packages/hicops/package.py]: I have adjusted the code to reflect this change.

File: var/spack/repos/builtin/packages/hicops/package.py (hunk 2 above)
Comment (NessieCanCode) [reply var/spack/repos/builtin/packages/hicops/package.py]: done

**Comment by NessieCanCode** [comment]
> I left an initial review with some required changes.

@alalazo, I believe I have addressed the changes in the initial review. I have re-requested a review.

**Comment by bernhardkaindl** [comment]
@NessieCanCode I saw this in the current package.py:
```
# TODO: Add timemory and mpip depends_on()
```
If that's not important now, yon can keep it as a reminder, but I wanted you to remind you to it.

**Review by bernhardkaindl — APPROVED** [review]
The standard variant passed a build on Ubuntu 18.05 with it's gcc-8.4.0

**Comment by alalazo** [comment]
@spackbot run pipeline

**Comment by bernhardkaindl** [comment]
> > I left an initial review with some required changes.
> 
> @alalazo, I believe I have addressed the changes in the initial review. I have re-requested a review.

@alalazo Ready to merge? I've tested that the new package builds just fine on Ubuntu 18.04 with gcc-8.4.0.

**Comment by NessieCanCode** [comment]
I'm going to close and resubmit this pull request, as it seems to be stalled

**Pull request closed without merging**